cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/CoreSchedulerService.java

```diff
@@ -431,6 +431,18 @@ public Void run(ProgramScheduleStoreDataset store, JobQueueDataset queue) {
   }
 
   @Override
+  public void modifySchedulesTriggeredByDeletedProgram(final ProgramId programId) {
+    checkStarted();
+    execute(new StoreAndQueueTxRunnable<Void, RuntimeException>() {
```

Comment:
ok, so this is where the transaction is, but shouldn't it retry if there is a transaction conflict?